### PR TITLE
Update Sort Tracking

### PR DESCRIPTION
**Pull request by theck13**
### Fix
Update the name of the analytics statistic for the sort order bar at the top of the note list updated in https://github.com/Automattic/simplenote-android/pull/1286 and add analytics for the sort order item in **_Settings_**.

### Test
Adding a breakpoint in the code can verify the event is triggered at the right time.  The steps below can be tested using a breakpoint [here](https://github.com/Automattic/simplenote-android/blob/c515a01897c094a0dfb21da0049ec93d4f86dbc1/Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java#L430) for the sort order bar and [here](https://github.com/Automattic/simplenote-android/blob/c515a01897c094a0dfb21da0049ec93d4f86dbc1/Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java#L209) for the sort order setting.

#### Bar
1. Tap sort bar.
2. Tap currently unselected sort option.
3. Notice breakpoint is triggered.
4. Tap sort bar.
5. Tap currently selected sort option.
6. Notice breakpoint is not triggered.

#### Setting
1. Open navigation drawer.
2. Tap ***Settings*** item in navigation drawer.
3. Tap ***Sort order*** item under ***Notes*** section.
4. Tap currently unselected sort option.
5. Notice breakpoint is triggered.
6. Tap ***Sort order*** item under ***Notes*** section.
7. Tap currently selected sort option.
8. Notice breakpoint is not triggered.

### Review
Only one developer is required to review these changes, but anyone can perform the review.

### Release
These changes do not require release notes.

### CODE REVIEW

File: Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java [hunk 1]
```diff
@@ -430,7 +430,7 @@ public boolean onMenuItemClick(MenuItem item) {
                         switch (item.getItemId()) {
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java:436]: Suggestion (💡): Consider extracting switch case duplicate code to prevent people from making update mistakes. For example:
```
switch (item.getItemId()) {
    case R.id.sort_alphabetical:
        updateSortOrder("alphabetical_az", ALPHABETICAL_ASCENDING);
        return true;
    case R.id.sort_alphabetical_reverse:
        updateSortOrder("alphabetical_za", ALPHABETICAL_DESCENDING);
        return true;
    case R.id.sort_newest_created:
        updateSortOrder("created_newest", DATE_CREATED_DESCENDING);
        return true;
    case R.id.sort_oldest_created:
        updateSortOrder("created_oldest", DATE_CREATED_ASCENDING);
        return true;
    case R.id.sort_newest_modified:
        updateSortOrder("modified_newest", DATE_MODIFIED_DESCENDING);
        return true;
    case R.id.sort_oldest_modified:
        updateSortOrder("modified_oldest", DATE_MODIFIED_ASCENDING);
        return true;
    default:
        return false;
}

private void updateSortOrder(String sortOrderLabel, int prefSortOrder) {
    AnalyticsTracker.track(
            AnalyticsTracker.Stat.LIST_SORTBAR_MODE_CHANGED,
            AnalyticsTracker.CATEGORY_SETTING,
            sortOrderLabel
    );
    mPreferences.edit().putString(PrefUtils.PREF_SORT_ORDER,
            String.valueOf(prefSortOrder)
    ).apply();
    refreshList();
}
```

PS: I understand that this is not part of the PR change, so please ignore my comment if you feel the same.

File: Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java [hunk 2]
```diff
@@ -206,6 +206,53 @@ public boolean onPreferenceChange(Preference preference, Object newValue) {
                 CharSequence[] entries = sortPreference.getEntries();
                 sortPreference.setSummary(entries[index]);
 
+                if (!sortPreference.getValue().equals(newValue)) {
+                    switch (index) {
```
Comment (ParaskP7) [inline Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java:222]: Suggestion (💡): Consider extracting switch case duplicate code to prevent people from making update mistakes. For example:
```
switch (index) {
    case 0:
        trackSortOrder("modified_newest");
        break;
    case 1:
        trackSortOrder("modified_oldest");
        break;
    case 2:
        trackSortOrder("created_newest");
        break;
    case 3:
        trackSortOrder("created_oldest");
        break;
    case 4:
        trackSortOrder("alphabetical_az");
        break;
    case 5:
        trackSortOrder("alphabetical_za");
        break;
}

private void trackSortOrder(String sortOrderLabel) {
    AnalyticsTracker.track(
            AnalyticsTracker.Stat.SETTINGS_SEARCH_SORT_MODE,
            AnalyticsTracker.CATEGORY_SETTING,
            sortOrderLabel
    );
}
```

**Review by ParaskP7 — APPROVED** [review]
👋 @theck13  !

I have reviewed and tested this PR as per the instruction, everything works as expected, good job! 🌟 

I have left a couple of suggestions (💡) for you to consider. I am going to approve this PR anyway, since none is blocking. I am **NOT** going to merge this PR yet to give you some time to apply any of my suggestions. However, feel free to ignore them and merge the PR yourself.

File: Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java (hunk 1 above)
Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java:436]: I added the `updateSortOrder` method in a23b117.

File: Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java (hunk 2 above)
Comment (theck13) [reply Simplenote/src/main/java/com/automattic/simplenote/PreferencesFragment.java:222]: I added the `trackSortOrder` method in 3fd33f8.

**Review by theck13 — COMMENTED** [review]
Thanks for the suggestions!  I added those changes.  I also added constants for the analytics labels in b85d912.

**Comment by ParaskP7** [comment]
👋 @theck13 !

Loved your changes, I reviewed them and LGTM! ❤️ 

Thank you so much for picking up my suggestions! 🙏
